### PR TITLE
Addition of Documentation for Utility Functions

### PR DESCRIPTION
**Pull request by festusdrakon**
Added descriptions to utility functions such as commonly used safe arithmetic operations, boolean, vector and statistical functions in response to issue #314

### CODE REVIEW

**Comment by chinmayshah99** [comment]
Can you fix the cpp style issues by running `make format-style-cpp`.

Also, can you change the PR title to be more descriptive?

**Comment by festusdrakon** [comment]
Edited the PR title to make it more descriptive and executed  `make format-style-cpp`  to rectify styling errors.

**Review by chinmayshah99 — APPROVED** [review]

**Review by alejandrosame — APPROVED** [review]
LGTM

**Comment by shaistha24** [comment]
Hi @festusdrakon suggestion - along with the description could you also add the fields or attributes needed for each and give a short intro of it too. Adding the direct link  to the source code is also great. Makes it easier for the new contributors to look it up on the source code. 

You may use this as an example- https://www.tensorflow.org/api_docs/python/tf/config/LogicalDevice

Other than that nice work @festusdrakon

File: src/bindings/PyDP/algorithms/util.cpp [hunk 1]
```diff
@@ -10,35 +10,63 @@ namespace dp = differential_privacy;
...
-    if (result) return k;
-    throw std::runtime_error("Result of subtraction will overflow.");
-  });
-  m.def("safe_square", [](int64_t i) {
-    int64_t k;
-    bool result = dp::SafeSquare(i, &k);
-    if (result) return k;
-    throw std::runtime_error("Result of squaring will overflow.");
-  });
+  m.def(
```
Comment (shaistha24) [inline src/bindings/PyDP/algorithms/util.cpp:13]: pls check my previous comment with the improvisations to be included.

**Review by shaistha24 — CHANGES_REQUESTED** [review]
Hi @festusdrakon   along with the description could you also add the fields or attributes needed for each and give a short intro of it too. Adding the direct link to the source code is also great. Makes it easier for the new contributors to look it up on the source code.

You may use this as an example- https://www.tensorflow.org/api_docs/python/tf/config/LogicalDevice

Other than that nice work @festusdrakon

**Comment by alejandrosame** [comment]
@festusdrakon  The failed test is still running against pipenv version of dev branch. In order to make github realise that the target branch has changed before rerunning the tests you can do one of the following:

- Pull the latest dev changes to your fork, merge them to your brannch and push the merged changes.

- Momentarily change the target branch to a different one, save and then change it back to dev (see https://stackoverflow.com/a/26986320).

Let me know if you need further clarification or guidance. While I think the first solution is the cleanest, I understand that the second one is just fast and straightforward :)

**Comment by chinmayshah99** [comment]
@festusdrakon can you please pull the latest changes from dev as @alejandrosame suggested?

**Comment by chinmayshah99** [comment]
#341 fixes this.

**Pull request closed without merging**